libs/tests/providerSelectionTest.js

```diff
@@ -0,0 +1,187 @@
+const assert = require('assert')
+const sinon = require('sinon')
+
+const ContractClient = require('../src/services/contracts/ContractClient')
+const Web3Manager = require('../src/services/web3Manager/index')
+const EthWeb3Manager = require('../src/services/ethWeb3Manager/index')
+const Web3 = require('web3')
+
+let contractClient
+
+function createWeb3Obj (host) {
+  return {
+    currentProvider: {
+      host
+    },
+    eth: {
+      Contract: () => {}
+    }
+  }
+}
+
+describe('Testing ContractClient class with ProviderSelection', () => {
+  beforeEach(async () => {
+    // Setting up ContractClient stub
+    const getRegistryAddressFn = () => { return '0xaaaaaaaaaaaaaaaaaaa' }
+    const web3Manager = new Web3Manager()
+    const web3obj = createWeb3Obj('https://audius.poa.network')
+    web3Manager.web3 = web3obj
+    web3Manager.web3Config = {}
+    web3Manager.web3Config.useExternalWeb3 = false
+    web3Manager.web3Config.internalWeb3Config = {}
+    const gateways = ['https://audius.poa.network', 'https://public.poa.network']
+    web3Manager.web3Config.internalWeb3Config.web3ProviderEndpoints = gateways
+    sinon.stub(web3Manager, 'provider').callsFake((arg1, arg2) => { return arg1 })
+    contractClient = new ContractClient(web3Manager, 'contractABI', 'contractRegistryKey', getRegistryAddressFn)
+  })
+
+  afterEach(async () => {
+    // Reset stub
+    sinon.restore()
+  })
+
+  /**
+     * Given: audius gateway is healthy
```

Comment:
nit, the * should be lined up with the previous line's star. does the linter not catch this?

Reply:
hm looks like my linter does allow this. i'll make the changes